_overviews/scala3-macros/tutorial/reflection.md

```diff
@@ -31,11 +41,126 @@ As it is a subtype of `Tree` we can also look into the `TreeMethods` to find mor
 Each type is also a module with some _static-ish_ methods, for example in the [`TypeReprModule`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TypeReprModule.html) we can find the method `TypeRepr.of[T]` which will create an instance of `Type` containing `T`.
 
 
-## Relation with expressions
-<!-- Term vs Expr -->
-<!-- Safty -->
-*Coming soon*
+## Relation with Expr/Type
+
+Expressions `Expr[T]` can be seen as wrapper around a `Term` that knows statically the type of the term.
+Bellow we use the `asTerm` extension method to transform the expression into a term.
+This extension method is only available after importing `quotes.reflect.asTerm`.
+Then we use `asExprOf[Int]` to transform the term back into `Expr[Int]`.
+This operation will fail if the term does not have provided type or if the term is not a valued expression.
+
+```scala
+def f(x: Expr[Int])(using Quotes): Expr[Int] =
+  import quotes.reflect.*
+  val tree: Term = x.asTerm
+  val expr: Expr[Int] = tree.asExprOf[Int]
+  expr
+```
+
+Similarly, we can also a `Type[T]` as a wrapper over a `TypeRepr` that knows statically its type.
+To get a `TypeRepr` we use `TypeRepr.of[X]` which expects a given `Type[X]` in scope (similar to `Type.of[X]`).
+We can also transform it back into a `Type[?]` using the `asType` method.
+As the type of `Type[?]` is not statically know we need to name it with an existential type to use it, can be done using the `'[t]` pattern.
```

Comment:
```suggestion
As the type of `Type[?]` is not statically know we need to name it with an existential type to use it. This can be achieved using a `'[t]` pattern.
```